core/src/main/java/feast/core/job/JobUpdateTask.java

```diff
@@ -73,147 +74,131 @@ public JobUpdateTask(
+      logAudit(Action.STATUS_CHANGE, job, auditMessage, runnerType, extId);
 
       return job;
     } catch (Exception e) {
       log.error(e.getMessage());
-      AuditLogger.log(
-          Resource.JOB,
-          jobId,
-          Action.STATUS_CHANGE,
-          "Job failed to be submitted to runner %s. Job status changed to ERROR.",
-          jobManager.getRunnerType().toString());
+      var auditMessage = "Job failed to be submitted to runner %s. Job status changed to ERROR.";
+      logAudit(Action.STATUS_CHANGE, job, auditMessage, runnerType);
 
       job.setStatus(JobStatus.ERROR);
       return job;
     }
   }
 
   /** Update the given job */
-  private Job updateJob(
-      Job job, List<FeatureSetProto.FeatureSet> featureSets, StoreProto.Store store) {
-    job.setFeatureSets(
-        featureSets.stream()
-            .map(
-                fs ->
-                    FeatureSet.fromProto(
-                        FeatureSetProto.FeatureSet.newBuilder()
-                            .setSpec(fs.getSpec())
-                            .setMeta(fs.getMeta())
-                            .build()))
-            .collect(Collectors.toList()));
-    job.setStore(feast.core.model.Store.fromProto(store));
-    AuditLogger.log(
-        Resource.JOB,
-        job.getId(),
-        Action.UPDATE,
-        "Updating job %s for runner %s",
-        job.getId(),
-        jobManager.getRunnerType().toString());
+  private Job updateJob(Job job) {
+    job.setFeatureSets(featureSetsFromProto(featureSets));
+    job.setStore(Store.fromProto(store));
+    logAudit(Action.UPDATE, job, "Updating job %s for runner %s", job.getId(), runnerType);
     return jobManager.updateJob(job);
   }
 
+  private Job updateStatus(Job job) {
+    JobStatus currentStatus = job.getStatus();
+    JobStatus newStatus = jobManager.getJobStatus(job);
+    if (newStatus != currentStatus) {
+      var auditMessage = "Job status updated: changed from %s to %s";
+      logAudit(Action.STATUS_CHANGE, job, auditMessage, currentStatus, newStatus);
+    }
+
+    job.setStatus(newStatus);
+    return job;
+  }
+
   String createJobId(String sourceId, String storeName) {
     String dateSuffix = String.valueOf(Instant.now().toEpochMilli());
     String sourceIdTrunc = sourceId.split("/")[0].toLowerCase();
     String jobId = String.format("%s-to-%s", sourceIdTrunc, storeName) + dateSuffix;
     return jobId.replaceAll("_", "-");
   }
+
+  // TODO: Either put in a util somewhere, or stop using proto types in domain logic altogether
+  private static List<FeatureSet> featureSetsFromProto(List<FeatureSetProto.FeatureSet> protos) {
+    return protos.stream()
+        .map(
+            fsp ->
+                FeatureSet.fromProto(
+                    FeatureSetProto.FeatureSet.newBuilder()
+                        .setSpec(fsp.getSpec())
+                        .setMeta(fsp.getMeta())
+                        .build()))
+        .collect(Collectors.toList());
```

Comment:
@zhilingc is the difference between calling:
1. ```
FeatureSet.fromProto(
                    FeatureSetProto.FeatureSet.newBuilder()
                        .setSpec(fsp.getSpec())
                        .setMeta(fsp.getMeta())
                        .build()))
```
and 2. `FeatureSet.fromProto(fsp)`?

If not we just do 2. and simplify the code.

Reply:
Right now it doesn't seem like there is any difference.

Reply:
No. I think this was leftover from when we were deliberating whether to pass only specs to the job (or the full proto object)